nostr/event.py

```diff
@@ -16,41 +18,65 @@ class EventKind(IntEnum):
     DELETE = 5
 
 
-class Event():
-    def __init__(
-            self, 
-            public_key: str, 
-            content: str, 
-            created_at: int = None, 
-            kind: int=EventKind.TEXT_NOTE, 
-            tags: "list[list[str]]"=[], 
-            id: str=None, 
-            signature: str=None) -> None:
-        if not isinstance(content, str):
+
+@dataclass
+class Event:
+    public_key: str = None
+    content: str = None
```

Comment:
Agreed. Moving it to first and removing the default value so that it becomes a required field.

Reply:
Actually strike making `content` a required field; breaks `EncryptedDirectMessage`.

I prefer defaulting to `None` in these cases because it signals a complete absence of any value being set yet. Kind of a holdover from database design, I guess (e.g. NULL says we never had their email addr; empty string says they requested to have it blanked).

Reply:
And... strike the "strike". I worked it out.

Sorry for thinking out loud (and back and forth) here!